example/lib/develop/repository/peripheral_list_repository.dart

```diff
@@ -0,0 +1,49 @@
+import 'dart:async';
+
+import 'package:blemulator_example/develop/model/ble_peripheral.dart';
+import 'package:blemulator_example/example_peripheral.dart';
+import 'package:flutter_ble_lib/flutter_ble_lib.dart';
+import 'package:blemulator/blemulator.dart';
+
+typedef void ScanEventOutputFunction(BlePeripheral peripheral);
+
+class PeripheralListRepository {
+  BleManager _bleManager;
+  StreamSubscription _scanEventsSubscription;
+
+  PeripheralListRepository(this._bleManager) {
+    // TODO: WIP - TEMPORARY PLACEMENT start
+    Blemulator().addSimulatedPeripheral(SensorTag());
+    Blemulator().addSimulatedPeripheral(SensorTag(id: "different id"));
+    Blemulator()
+        .addSimulatedPeripheral(SensorTag(id: "yet another different id"));
+    Blemulator().simulate();
+    // TODO: WIP - TEMPORARY PLACEMENT end
+
+    _bleManager.createClient();
+  }
+
+  void startPeripheralScan({ScanEventOutputFunction scanEventOutput}) {
```

Comment:
why is it optional and what happens when it is null?

Reply:
Nothing happens when event is optional - bloc does not receive any scan events then when they arrive at the repository. 

It might be worth changing to a non-optional though (I only thought that in some potential cases when someone wants to reuse repository somewhere, they might not be interested in receiving scanResult callbacks).